objects.py

```diff
@@ -3,38 +3,48 @@
     Edit the class definitions to make the tests pass.
 """
 
-
 class Dessert(object):
 
-    def __init__():
+    def __init__(self, price, calories=None):
         # Edit me!
         # You need to be able to initialize a Dessert object with arguments:
         # price - required
         # calories - optional
-
+        self.price = price
+        self.calories = calories
         # This should set the object's price and calories, accessible by
         # .price and .calories respectively.
         pass
 
     # Add a calories_per_dollar method that returns the calories per dollar
     # for the dessert.
+    def calories_per_dollar(self):
+        calories_per_dollar = self.calories/self.price
+        print calories_per_dollar
 
     # Define a method is_a_cake on Dessert that returns False
-
+    def is_a_cake(self):
+        return False
+    
 
 class Cake(Dessert):
 
-    def __init__():
+    def __init__(self, kind):
         # Edit me!
         # Cakes all cost the same amount and have the same calories, so their
         # price and calories can be set at the class-level, not during init.
         # However, we need to be able to tell cakes apart. Accept argument:
         # kind - required
+        super(Cake, self).__init__(price,calories)
```

Comment:
This probably causes an error because at this point, `price` and `calories` aren't defined. You also don't need to call the `super` method. Will walk through the solution in class!